extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -211,6 +205,28 @@ class AmpYoutube extends AMP.BaseElement {
     }
   }
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    if (mutations['data-videoid'] !== undefined) {
+      this.videoid_ = this.getVideoId_();
+      if (this.iframe_) { // `null` if element hasn't been laid out yet.
```

Comment:
no need to refresh the iframe, YouTube supports dynamically loading a video which should be much faster than a full reload, the following should work (not tested though)

```
    this.playerReadyPromise_.then(() => {
      this.sendCommand_('loadVideoById', this.videoid_);
    });
```